app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
```

Comment:
```suggestion
  before_save{email.downcase!}
```
so sánh downcase! và downcase nhé

Reply:
downcase và downcase! đều trả về chuỗi chữ thường, nếu chuỗi không có sự thay đổi thì downcase! trả về nil còn downcase vẫn trả về chuỗi như ban đầu ạ.

Reply:
vẫn thiếu, thêm gì nữa không e ?

Reply:
downcase trả về chuỗi thường nhưng không làm thay đổi chuỗi ban đầu còn downcase! thì có a.